### PR TITLE
Add check for Union[Match, ...] or Union[Pattern, ...]

### LINKED ISSUE #6: python3.5.[01]: Union[Pattern, str] raises an error
Need to make sure this gets quoted instead of left unquoted:

```pycon
>>> Union[AnyStr, str]
typing.Union[~AnyStr, str]
>>> Union[str, Pattern]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/typing.py", line 534, in __getitem__
    dict(self.__dict__), parameters, _root=True)
  File "/usr/local/lib/python3.5/typing.py", line 491, in __new__
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
  File "/usr/local/lib/python3.5/typing.py", line 491, in <genexpr>
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
TypeError: issubclass() arg 1 must be a class
```

### PR DESCRIPTION
**Pull request by atugushev**
Resolves #6.

### CODE REVIEW

File: flake8_typing_imports.py [hunk 1]
```diff
@@ -447,18 +449,52 @@ def __init__(self) -> None:
...
+            node.id in self.from_imported_names
+        ) or (
+            isinstance(node, ast.Attribute) and
+            isinstance(node.value, ast.Name) and
+            node.value.id == 'typing' and
+            node.attr in names
+        )
 
     def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
-        if node.module == 'typing' and self._level == 0:
```
Comment (asottile) [inline flake8_typing_imports.py]: I think you can keep this logic

File: flake8_typing_imports.py [hunk 2]
```diff
@@ -447,18 +449,52 @@ def __init__(self) -> None:
         self.imports: Dict[str, List[Tuple[int, int]]]
         self.imports = collections.defaultdict(list)
         self.defined_overload = False
+        self.unions_pattern_or_match: List[Tuple[int, int]] = []
+        self.from_imported_names: Set[str] = set()
+
+    def _is_typing(self, node: ast.AST, names: Container[str]) -> bool:
```
Comment (asottile) [inline flake8_typing_imports.py]: `Tuple[str, ...]`

File: flake8_typing_imports.py [hunk 3]
```diff
@@ -540,3 +576,14 @@ def run(self) -> Generator[Tuple[int, int, str, Type[Any]], None, None]:
         ):
             for line, col in visitor.imports['overload']:
                 yield line, col, msg, type(self)
+
+        msg = (
+            'TYP003 Union[Match, ...] or Union[Pattern, ...] '
+            'must be quoted <3.5.2'
```
Comment (asottile) [inline flake8_typing_imports.py]: must be quoted in <3.5.2

File: tests/flake8_typing_imports_test.py [hunk 4]
```diff
@@ -118,3 +118,76 @@ def test_missing_overload_define():
         )
     with version_ctx(Version(3, 5, 2)):
         assert not results('from typing import overload')
+
+
+@pytest.mark.parametrize(
+    's', [
```
Comment (asottile) [inline tests/flake8_typing_imports_test.py]: prefer tuples to lists when equivalent: https://github.com/asottile/scratch/wiki/Prefer-tuples-to-lists

File: tests/flake8_typing_imports_test.py [hunk 5]
```diff
@@ -118,3 +118,76 @@ def test_missing_overload_define():
...
+        ),
+        pytest.param(
+            'from typing import Match, Union\n'
+            'def foo(bar: "Union[Match, str]"): pass\n',
+            id='quoted Match',
+        ),
+        pytest.param(
+            'from typing import Pattern as Pttrn, Union\n'
+            'def foo(bar: Union[Pttrn, str]): pass\n',
+            id='Pattern imported as Pttrn',
```
Comment (asottile) [inline tests/flake8_typing_imports_test.py]: in an ideal world we'd detect this as well, but it's impossible to handle _every_ case of this -- also it's a bit nonsensical

Comment (atugushev) [reply tests/flake8_typing_imports_test.py]: Should I remove this case?

Comment (asottile) [reply tests/flake8_typing_imports_test.py]: maybe just a comment `# this is broken too, but so unlikely we elect not to detect it`

Comment (atugushev) [reply tests/flake8_typing_imports_test.py]: Or move to the above test and mark it xfail?

Comment (asottile) [reply tests/flake8_typing_imports_test.py]: up to you, idrc

File: flake8_typing_imports.py (hunk 1 above)
Comment (atugushev) [reply flake8_typing_imports.py]: Do you mean it should be changed to this?
```python
if node.module == 'typing' and self._level == 0:
    for name in node.names:
        self.imports[name.name].append((node.lineno, node.col_offset))
        if not name.asname:
            self.from_imported_names.add(name.name)

```

Comment (asottile) [reply flake8_typing_imports.py]: yeah

**Comment by atugushev** [comment]
Thanks for reviewing this! Addressed in 75943c1f84f70d7ae12b607e7de38f1d6484f5fd.

**Comment by atugushev** [comment]
Thanks!

**Pull request merged**